TESTING.md

```diff
@@ -1,6 +1,151 @@
-# VSCode testing
+# ACPaaS UI Testing Guide
 
-To make life easier in VScode for testing with Jest.
-We suggest you install the following plugin:
+When developing, testing is a major part of the work.
+We use [Jest](https://jestjs.io) in combination with [Enzyme](https://airbnb.io/enzyme) as our test suite.
 
-https://github.com/jest-community/vscode-jest
\ No newline at end of file
+To make life easier in Visual Studio Code for testing with Jest, we suggest that you install [this VS Code Jest extension](https://github.com/jest-community/vscode-jest).
+
+## ACPaaS UI Test Automation Guide
+
+When automating tests, one of the challenges is to keep the test resilient to code changes.
+Using [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) can help you with that, by creating unique selectors for your tests.
+
+### Using unique data attributes with ACPaaS UI
+
+It's not possible to add unique attributes to the ACPaaS UI components for the simple reason that the attribute won't be unique anymore when you use the component multiple times on the same page.
+
+We solve this problem by adding unique attributes to the individual ACPaaS UI components at app level.
+
+#### Example
+
+The following example shows the ACPaaS UI Accordion with the unique `qa` data attribute `id-1234`.
+
+```html
+<Accordion qa="id-1234"/>
+    <AccordionTab />
+    <AccordionTab />
+</Accordion>
+```
+
+Which will result in:
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  ...
+</div>
+```
+
+### Targeting nested elements
+
+If you want to target a specific element within the component, use their CSS class in combination with the unique data attribute of the parent component.
+
+#### Example
+
+In the following example the accordion tab can be opened or closed by toggling the `<div>` tag with the CSS class `m-accordion__tab`, which is nested in the accordion with data attribute `data-qa="id-1234"`.
+So the compiled code would look like this:
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  <div ... class="m-accordion__tab">
+  ...
+</div>
+```
+
+#### Overview
```

Comment:
Sorry, missed this: this heading should be `###`

Reply:
Done.